src/Microsoft.AspNetCore.Mvc/DesignTimeMvcServiceCollectionProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class DesignTimeMvcServiceCollectionProvider
+    {
+        public static IServiceCollection PopulateServiceCollection(IServiceCollection services, string assemblyName)
```

Comment:
I wonder if we should have additional options to set content root and application root.